server/src/internalClusterTest/java/org/elasticsearch/snapshots/DataStreamsSnapshotsIT.java

```diff
@@ -0,0 +1,224 @@
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.admin.cluster.snapshots.create.CreateSnapshotResponse;
+import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.elasticsearch.action.admin.cluster.snapshots.restore.RestoreSnapshotResponse;
+import org.elasticsearch.action.admin.indices.datastream.CreateDataStreamAction;
+import org.elasticsearch.action.admin.indices.datastream.DeleteDataStreamAction;
+import org.elasticsearch.action.admin.indices.datastream.GetDataStreamAction;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.indices.DataStreamIT;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.transport.RemoteTransportException;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class DataStreamsSnapshotsIT extends AbstractSnapshotIntegTestCase {
+
+    public void testSnapshotAndRestore() throws Exception {
+        Client client = client();
+
+        Path location = randomRepoPath();
+        createRepository("repo", "fs", location);
+
+        DataStreamIT.createIndexTemplate("t1", "@timestamp", "ds", "other-ds");
+
+        CreateDataStreamAction.Request request = new CreateDataStreamAction.Request("ds");
+        AcknowledgedResponse response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        request = new CreateDataStreamAction.Request("other-ds");
+        response = client.admin().indices().createDataStream(request).get();
+        assertTrue(response.isAcknowledged());
+
+        Map<String, Integer> source = Collections.singletonMap("@timestamp", 123);
+        IndexResponse indexResponse = client.prepareIndex("ds")
+            .setOpType(DocWriteRequest.OpType.CREATE)
+            .setSource(source)
+            .get();
+        assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster()
+            .prepareCreateSnapshot("repo", "snap")
+            .setWaitForCompletion(true)
+            .setIndices("ds")
+            .setIncludeGlobalState(false)
+            .get();
+
+        RestStatus status = createSnapshotResponse.getSnapshotInfo().status();
+        assertEquals(RestStatus.OK, status);
+
+        GetSnapshotsResponse snapshot = client.admin().cluster().prepareGetSnapshots("repo").setSnapshots("snap").get();
+        List<SnapshotInfo> snap = snapshot.getSnapshots("repo");
+        assertEquals(1, snap.size());
+        assertEquals(Collections.singletonList(".ds-ds-000001"), snap.get(0).indices());
```

Comment:
Minor, but having gone through the exercise of changing the backing index naming scheme, it would be nice to use `DataStream::getDefaultBackingIndexName` here so any future changes are automatically picked up.

Reply:
Good suggestion, I've changed the code accordingly